pennylane/variable.py

```diff
@@ -132,10 +144,35 @@ def val(self):
             float: current value of the Variable
         """
         # pylint: disable=unsubscriptable-object
-        if self.name is None:
+        if not self.is_kwarg:
             # The variable is a placeholder for a positional argument
             return Variable.free_param_values[self.idx] * self.mult
 
         # The variable is a placeholder for a keyword argument
         values = Variable.kwarg_values[self.name]
         return values[self.idx] * self.mult
+
+    def render(self, show_name_only=False):
+        """Returns a string representation of the Variable.
+
+        Args:
+            show_name_only (bool, optional): Render the name instead of the value.
+
+        Returns:
+            str: A string representation of the Variable
+        """
+        if not show_name_only:
+            if self.is_kwarg and Variable.kwarg_values and self.name in Variable.kwarg_values:
+                return str(round(self.val, 3))
+
+            if (
+                not self.is_kwarg
+                and Variable.free_param_values is not None
+                and len(Variable.free_param_values) > self.idx
+            ):
+                return str(round(self.val, 3))
+
+        if self.mult != 1:
+            return "{}*{}".format(str(round(self.mult, 3)), self.name)
+
+        return self.name
```

Comment:
Should this be `__str__`, and the existing `__str__` becoming `__repr__`?

Reply:
I have no hard feelings concerning this. But if the above thing would become `__str__` you would essentially get `w[0, 3]` instead of `Variable 0: name = w[0, 3]`. I guess it is not detrimental to show more info in `str` (as is the current approach)